### ISSUE TITLE
allow validating of config json before starting consul agent

### ISSUE THREAD

**Issue by mtougeron** [hashicorp/consul#334]
If you have invalid json in your configuration consul agent still succeeds but then later crashes. It would be nice to have an option to not start/reload unless the json is well-formed/valid.

We are running v0.3.1

**Comment by ryanuber** [comment]
#576 added more strict config checking. The JSON syntax is also checked in the parser.
